sdk/core/azure-core/inc/azure/core/paged_response.hpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Provides support for paged responses.
+ */
+
+#pragma once
+
+#include <cstdlib>
+#include <string>
+
+#include "azure/core/context.hpp"
+#include "azure/core/http/raw_response.hpp"
+#include "azure/core/nullable.hpp"
+
+namespace Azure { namespace Core {
+
+  /**
+   * @brief Defines the base type and behavior for a paged response.
+   *
+   * @remark The template is using for making static-inheritance.
+   *
+   * @remark T classes must implement the way to get and move to the next page.
+   *
+   * @tparam T A class type for static-inheritance.
+   */
+  template <class T> class PagedResponse {
+  private:
+    // The field used to check when the end of the response is reached. All responses from a service
+    // will always come with a payload that represents a page. The page might or might not contain
+    // elements in the page. `m_hasPage` is then turned to `false` once `MoveToNextPage` is called
+    // on the last page.
+    bool m_hasPage = true;
```

Comment:
Great comment. Maybe add, we default it to true because...

```suggestion
    // The field used to check when the end of the response is reached. We default it true as the starting point because all responses from a service
    // will always come with a payload that represents at least one page. The page might or might not contain
    // elements in the page. `m_hasPage` is then turned to `false` once `MoveToNextPage` is called
    // on the last page.
    bool m_hasPage = true;
```